File: easy/121 (Best time to buy and sell stocks)/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int ans = 0; 
        int mn = prices[0];
        for(int i = 1; i < prices.length; i++){
            ans = Math.max(ans, (prices[i] - mn));
            mn = Math.min(mn,prices[i]);
        }
            return ans;
    }
}

File: easy/121 (Best time to buy and sell stocks)/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        mn = prices[0]
        for i in range(1 , len(prices)):
            ans = max(ans, prices[i] - mn)
            mn = min(mn,prices[i])
        return ans